packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -38,28 +55,65 @@ class AppBar extends StatelessWidget {
     assert((tabBar != null) ? flexibleSpace == null : true);
   }
 
+  /// A widget to display before the [title].
+  ///
+  /// If this field is null and this app bar is used in a [Scaffold], the
+  /// [Scaffold] will fill this field with an appropriate widget. For example,
+  /// if the [Scaffold] also has a [Drawer], the [Scaffold] will fill this
+  /// widget with an [IconButton] that opens the drawer. If there's no [Drawer]
+  /// and the parent [Navigator] can go back, the [Scaffold] will fill this
+  /// field with an [IconButton] that calls [Navigator.pop].
   final Widget leading;
+
+  /// The primary widget displayed in the app bar.
+  ///
+  /// Typically a [Text] widget containing a description of the current contents
+  /// of the app.
   final Widget title;
+
+  /// Widgets to display after the title.
+  ///
+  /// Typically these widgets are [IconButton]s representing common operations.
+  /// For less common operations, consider using a [PopupMenuButton] as one of
+  /// the actions.
   final List<Widget> actions;
-  final WidgetBuilder flexibleSpace;
-  final double foregroundOpacity;
+
+  /// A widget to aid in building scroll-based collapsing app bar effects.
+  ///
+  /// Typically a [FlexibleSpaceBar]. See [FlexibleSpaceBar] for details.
+  final Widget flexibleSpace;
+
+  /// A horizontal strip of tabs to display at the bottom of the app bar.
   final TabBar<dynamic> tabBar;
+
+  /// The height at which to place this app bar.
```

Comment:
we use height for vertical extent. it's mildly confusing to use it for the z axis here.